File: javascript/space-age/space-age.js
class SpaceAge {
	constructor(earthSec) {
		this.seconds = earthSec;
		this.earthYear = this.seconds / 31557600;
		this.t = {
			Mercury: 0.2408467,
			Venus: 0.61519726,
			Mars: 1.8808158,
			Jupiter: 11.862615,
			Saturn: 29.447498,
			Uranus: 84.016846,
			Neptune: 164.79132
		};
	}

	onEarth() {
		return Number(this.earthYear.toFixed(2));
	}

	onMercury() {
		return Number((this.earthYear / this.t.Mercury).toFixed(2));
	}

	onVenus() {
		return Number((this.earthYear / this.t.Venus).toFixed(2));
	}

	onMars() {
		return Number((this.earthYear / this.t.Mars).toFixed(2));
	}

	onJupiter() {
		return Number((this.earthYear / this.t.Jupiter).toFixed(2));
	}

	onSaturn() {
		return Number((this.earthYear / this.t.Saturn).toFixed(2));
	}

	onUranus() {
		return Number((this.earthYear / this.t.Uranus).toFixed(2));
	}

	onNeptune() {
		return Number((this.earthYear / this.t.Neptune).toFixed(2));
	}
}
 
module.exports = SpaceAge

File: go/space-age/space_age.go
// Given an age in seconds, calculate how old someone would be on other planets
package space

type Planet string

var spaceAge = map[Planet]float64{
	"Mercury": 0.2408467,
	"Venus":   0.61519726,
	"Mars":    1.8808158,
	"Jupiter": 11.862615,
	"Saturn":  29.447498,
	"Uranus":  84.016846,
	"Neptune": 164.79132,
	"Earth":   1,
}

// Function Age return age in years on another planet
func Age(seconds float64, planet Planet) float64 {
	var earthSeconds float64 = 31557600
	earthYear := seconds / earthSeconds
	return earthYear / spaceAge[planet]
}
